regress/pesterTests/TestUtils.ps1

```diff
@@ -0,0 +1,129 @@
+﻿If ($PSVersiontable.PSVersion.Major -le 2) {$PSScriptRoot = Split-Path -Parent $MyInvocation.MyCommand.Path}
+Import-Module $PSScriptRoot\..\..\contrib\win32\openssh\OpenSSHUtils.psd1 -Force
+$Script:newProcesses = @()
+$Script:OpenSSHBinPath=""
+
+function Get-OpenSSHBinPath
+{
+    param([string]$Configuration = "Release")
+
+    [string] $NativeHostArch = $env:PROCESSOR_ARCHITECTURE
+    if($NativeHostArch -eq 'x86')
+    {
+        $NativeHostArch = "Win32"
+    }
+    else
+    {
+        $NativeHostArch = "x64"
+    }  
+    
+
+    $BinPath = Resolve-Path (join-path "..\..\bin\$NativeHostArch" "$Configuration\sshd.exe") -ErrorAction Ignore
+    if($BinPath -eq $null)
+    {
+        $BinPath = get-command sshd.exe -ErrorAction SilentlyContinue 
+    }
+    
+    if($BinPath -eq $null)
+    {
+        Throw "Cannot find sshd.exe. Please build openssh in repro or set the Path environment to openssh daemon."
+    }
+
+    $SSHDBinPath = $BinPath.Path
+
+    $Script:OpenSSHBinPath = Split-Path $SSHDBinPath
+    $Script:OpenSSHBinPath
+}
+
+function Start-SSHDDaemon
+{
+    param(
+    [parameter(Mandatory=$true)]
+    [string]$SSHD_Config_Path)    
+    
+    $Script:newProcesses = $null
+    $existingProcesseIDs = @()
+    if(($existingProcesses = Get-Process -name sshd -ErrorAction SilentlyContinue)){
+        $existingProcesseIDs = $existingProcesses.id
+    }    
+    
+    Start-process -FilePath "$($Script:OpenSSHBinPath)\sshd.exe" -ArgumentList "-f $SSHD_Config_Path" -NoNewWindow
+    #Start-Process -FilePath "$Script:OpenSSHBinPath\sshd.exe" -ArgumentList @("-ddd", "-f $SSHD_Config_Path")
+    
+    #Sleep for 1 seconds for process to ready to listener
+    $num = 0
+    do
+    {
+        $Script:newProcesses = Get-Process -name sshd -ErrorAction SilentlyContinue | Where-Object {$_.id -notin $existingProcesseIDs}
+        start-sleep 1
+        $num++
+        if($num -gt 30) { break }
+    } while ($Script:newProcesses -eq $null)    
+}
+
+function Stop-SSHDDaemon
+{
+    if($Script:newProcesses -and $Script:newProcesses.count -gt 0) {
+        $Script:newProcesses | Stop-Process -ErrorAction SilentlyContinue
+     }
+     $Script:newProcesses = $null
+ 
+}
+
+function Write-SSHDConfig
+{
+    param(
+        $Port = 47002,
```

Comment:
Since probably 75% of the tests use the same setup, we could get rid of this parametrized version and make it part of common setup.

Reply:
ok